interface/java_binding/src/main/java/org/khronos/ktx/KtxBasisParams.java

```diff
@@ -410,29 +412,36 @@ public char[] getInputSwizzle() {
 	/**
 	 * Set the swizzle that should be applied to the input.<br>
 	 * <br>
-	 * This swizzle must match the regular expression /^[rgba01]{4}$/.<br>
+	 * When the given swizzle is <code>null</code> or all its elements are
+	 * <code>0</code>, then no swizzling will be applied to the input.<br>
+	 * <br>
+	 * Otherwise, this swizzle must match the regular expression 
+	 * <code>/^[rgba01]{4}$/</code>.<br>
 	 * <br>
-	 * When the given swizzle is <code>null</code>, then no swizzling will be
-	 * applied to the input.
 	 *
 	 * @param inputSwizzle The swizzle
 	 */
 	public void setInputSwizzle(char[] inputSwizzle) {
-		if (inputSwizzle == null) {
-			this.inputSwizzle = new char[4];
-			return;
-		}
-		if (inputSwizzle.length != 4) {
-			throw new IllegalArgumentException("The inputSwizzle must contain 4 characters");
-		}
-		String valid = "rgba01";
-		for (int i = 0; i < inputSwizzle.length; i++) {
-			char c = inputSwizzle[i];
-			if (valid.indexOf(c) == -1) {
-				throw new IllegalArgumentException("The inputSwizzle may only consist of 'rgba01', but contains " + c);
-			}
-		}
+	    char defaultSwizzle[] = new char[4];
```

Comment:
Same question as for the AstcParams case.

Reply:
Same answer as above. 

With the addendum: I also didn't like the duplication, and considered to pull out this block into a function, so that the 'setter' could boil down to something like
`this.inputSwizzle = Swizzling.validate(inputSwizzle);`
This would also allow to avoid the duplication of the unit tests (which may be a stronger case to make). 

I somehow hestiate(d) to create such a `class Swizzling` just for this purpose, but will have another look: There might be other suplicated blocks that could justify moving this into some `class Utils` that does some of the parameter checks.

Reply:
So I think that, in this case, it makes sense to move this to a common function. So I just did that, and there now is a `KtxUtilities.validateSwizzle` function. This avoids the duplication in the `BasisParams`/`AstcParams` themself, _and_ it avoids the duplication of the unit tests.

Reply:
> This avoids the duplication in the BasisParams/AstcParams themself, and it avoids the duplication of the unit tests.

Sounds good. I'll be away the next 3 days. I'll review again on my return.